vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -11,34 +10,177 @@
+  winrt::Microsoft::ReactNative::CompositionHwndHost host{nullptr};
+
+  if (data) {
+    winrt::check_hresult(data->QueryInterface(
+        winrt::guid_of<winrt::Microsoft::ReactNative::CompositionHwndHost>(),
+        winrt::put_abi(host))); // look into the data for a CompositionHwndHost and store it in host
+    auto result = static_cast<LRESULT>(host.TranslateMessage(message, wparam, lparam));
+    if (result) {
+      return result;
+    }
+  }
+
+  switch (message) {
+    case WM_NCCREATE: { // sent before WM_CREATE, lparam should be identical to members of CreateWindowEx
+      auto createStruct = reinterpret_cast<CREATESTRUCT *>(lparam); // CreateStruct
+      auto windowData = static_cast<::IUnknown *>(createStruct->lpCreateParams);
+      SetProp(hwnd, CompHostProperty, reinterpret_cast<::IUnknown *>(windowData)); // adds new properties to window
+      break;
+    }
+    case WM_CREATE: { // recieves after window is created but before visible
+      // host.Initialize((uint64_t)hwnd); cause Modal to throw a not registered error
+      break;
+    }
+    case WM_CLOSE: {
+      // Just hide the window instead of destroying it
+      ::ShowWindow(hwnd, SW_HIDE);
+      return 0;
+    }
+    case WM_DESTROY: { // called when we want to destroy the window
+      data->Release();
+      SetProp(hwnd, CompHostProperty, nullptr);
+      break;
+    }
+  }
+
+  return DefWindowProc(hwnd, message, wparam, lparam);
+}
+
+// Creates and Register a new window class
+void WindowsModalHostComponentView::RegisterWndClass() noexcept {
+  static bool registered = false;
+  if (registered) {
+    return;
+  }
+
+  HINSTANCE hInstance =
+      GetModuleHandle(NULL); // returns a handle to the file used to create the calling process (.exe file)
+
+  WNDCLASSEX wcex = {}; // contains window class information
+  wcex.cbSize = sizeof(wcex); // size of windows class (bytes)
+  wcex.style = CS_HREDRAW | CS_VREDRAW; // class style (redraw window on size adjustment)
+  wcex.lpfnWndProc = &ModalBoxWndProc; // pointer to windows procedure
+  wcex.cbClsExtra = DLGWINDOWEXTRA; // extra bytes to allocate
+  wcex.cbWndExtra =
+      sizeof(winrt::impl::abi<winrt::Microsoft::ReactNative::ICompositionHwndHost>::type *); // extra bytes to allocate
+  wcex.hInstance = hInstance;
+  wcex.hCursor = LoadCursor(nullptr, IDC_ARROW); // handle to class cursor
+  wcex.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1); // background color
+  wcex.lpszClassName = c_modalWindowClassName; // specify resource name
+  ATOM classId = RegisterClassEx(&wcex); // register new windows class
+  WINRT_VERIFY(classId); // 0 = fail
+  winrt::check_win32(!classId);
+
+  registered = true;
+}
+
 void WindowsModalHostComponentView::MountChildComponentView(
     const winrt::Microsoft::ReactNative::ComponentView &childComponentView,
     uint32_t index) noexcept {
-  assert(false);
+  // assert(false);
```

Comment:
Why are these asserts disabled?

Reply:
It throws an exception when opening a new hwnd! Right now it's just commented out for me to look at in a later PR :)

Reply:
Cool. Leave a comment that makes it clear?
```suggestion
  // Disabled due to partial Modal implementation. Tracking re-enablement with task list here: https://github.com/microsoft/react-native-windows/issues/11157
  // assert(false);
```